Replace `isexported` with `ispublic` as criterion for `@autodocs` (#2629)
JuliaDocs/Documenter.jl: CHANGELOG.md

* Symlinks are now followed when walking the docs directory. ([#2610])
* PDF/LaTeX builds now throw a more informative error when `sitename` is not provided. ([#2636])
* `@autodocs` now lists public unexported symbols by default (i.e. when `Public = true`). ([#2629])

  This is **potentially breaking** as it can cause previously working builds to fail if they are being run in strict mode.
  This can happen if there are unexported docstring marked with `public` that are being included manually with e.g. `@docs` blocks, and there is also an `@autodocs` block including all public docstrings.
  The solution is to remove the duplicate inclusion.

> * `checkdocs` has a new option `:public` to check for unincluded public symbols that are marked with `public`, but are unexported. ([#2629])

Author: ```suggestion
* `checkdocs` has a new option `:public` to check that unexported symbols marked with `public` are included in the docs. ([#2629])
```
